### PR TITLE
more cleanup in debug_native - linux

### PR DESCRIPTION
**Pull request by alvarofe**
I've not tried on Android since I don't have any android device. It could fail to compile since I don't know if the ostype is gnulinux or android. It's a quick fix if that is wrong

```
+ifeq ($(OSTYPE),gnulinux)
+OBJS+= p/${NATIVE_OBJS}
+endif
```

Also I've tested on openbsd. But at least in my case I've observed that when doing ./configure it setups ostype to gnulinux instead of openbsd so in order to work it must be executed`./configure --with-ostype=openbsd` (I need to test more in openbsd because I don't have procfs mounted and it fails in some operations, but I think that is the normal since with master I get the same but only to be sure that I didn't break nothing ). I think the issue reside here in the configure file. @radare should know if this is wrong or what since I am not a big expert on make xD 

```
if [ "$USEROSTYPE" = "auto" ]; then
if [ "$HOST_OS" = "mingw32_nt" ]; then
USEROSTYPE="windows"
else
if [ "$HOST_OS" = "linux" ]; then
USEROSTYPE="gnulinux"
else
if [ "$HOST_OS" = "gnu" ]; then
USEROSTYPE="gnulinux"
HAVE_LIB_DL="1"
DL_LIBS="-ldl"
else
if [ "$HOST_OS" = "gnu/kfreebsd" ]; then
USEROSTYPE="gnulinux"
else
if [ "$HOST_OS" = "netbsd" ]; then
USEROSTYPE="gnulinux"
else
if [ "$HOST_OS" = "freebsd" ]; then
USEROSTYPE="gnulinux"
else
if [ "$HOST_OS" = "openbsd" ]; then
USEROSTYPE="gnulinux"
else

```

If I have some issues with indentation just highlight them because I don't know why I get different outputs from git diff, github desktop app and vim.

Thanks and let me know if I broke something :)

### CODE REVIEW

**Comment by radare** [comment]
The Android build fails: you can test this by running `sys/android-arm.sh`

```
p/debug_native.c:1031: error: undefined reference to 'linux_desc_list'
p/debug_native.c:79: error: undefined reference to 'linux_reg_profile'
p/debug_native.c:559: error: undefined reference to 'linux_reg_write'
p/debug_native.c:504: error: undefined reference to 'linux_reg_read'
p/debug_native.c:90: error: undefined reference to 'linux_handle_signals'
p/debug_native.c:123: error: undefined reference to 'linux_step'
p/debug_native.c:132: error: undefined reference to 'linux_attach'
p/debug_native.c:343: error: undefined reference to 'linux_info'
p/debug_native.c:359: error: undefined reference to 'linux_thread_list'
collect2: error: ld returned 1 exit status
../rules.mk:66: recipe for target 'rabin2' failed
make[2]: *** [rabin2] Error 1
```

**Comment by radare** [comment]
About the OSTYPE. im using gnulinux as a generalization for free unixes, which tend to be pretty common, the include files will already identify the OS properly. but that's true that maybe we should split this to support each OS independently, instead of generalizing

**Comment by radare** [comment]
Also, note that all C files must contain a header with license and author or contributors

**Comment by radare** [comment]
Note, that i pushed some changes that may break your pr . hopefully not much O:)

File: libr/debug/p/debug_native.c [hunk 1]
```diff
@@ -98,17 +65,34 @@ static int r_debug_handle_signals (RDebug *dbg) {
 /* begin of debugger code */
 #if DEBUGGER
 
-//this is temporal
-#if __APPLE__
+static int r_debug_native_continue(RDebug *dbg, int pid, int tid, int sig);
```
Comment (ret2libc) [inline libr/debug/p/debug_native.c]: space after function name. same for other lines

File: libr/debug/p/debug_native.c [hunk 2]
```diff
@@ -155,32 +129,18 @@ static int r_debug_native_step(RDebug *dbg) {
...
 #endif
 	if (pid == dbg->pid)
 		return pid;
 #if __WINDOWS__ && !__CYGWIN__
 	HANDLE process = w32_open_process (PROCESS_ALL_ACCESS, FALSE, pid);
-	if (process != (HANDLE)NULL && DebugActiveProcess (pid))
+	if (process != (HANDLE)NULL && DebugActiveProcess (pid)) {
 		ret = w32_first_thread (pid);
-	else ret = -1;
+	}
```
Comment (ret2libc) [inline libr/debug/p/debug_native.c]: else on the same line as the closing brace

**Comment by alvarofe** [comment]
I close the PR since there are conflicts with master but I think not much. I will take into account all the comments :)

**Pull request closed without merging**